packages/components/src/components/numpad/numpad.tsx

```diff
@@ -89,28 +90,28 @@ const Numpad = ({
                 }
                 setFloat(true);
                 if (default_value.toString().length === 0) {
-                    updateValue(concatenate(num, 0));
+                    updateValue(concatenate(num, '0'));
                 } else {
                     updateValue(concatenate(num, default_value));
                 }
 
                 break;
             default:
                 if (String(default_value) === '0') {
-                    updateValue(concatenate(String(num), 0));
+                    updateValue(concatenate(num, ''));
                 } else {
-                    const regex = /(?:^\d{0,35}\.)(\d{0,35})$/;
-                    const matches = regex.exec(String(default_value));
+                    const regex = /(?:\d+\.)(\d+)$/;
```

Comment:
Hi @niloofar-deriv , with this Regex, u r eliminating the max 35 number of digits. Is that expected?

Reply:
Hi @likhith-deriv revert this could u please check it again :pray: